src/app_engine/analytics.py

```diff
@@ -26,39 +21,22 @@ class Analytics(object):
 
   """
   def __init__(self):
-    is_running_locally = os.environ.get('APPLICATION_ID', '').startswith('dev')
-
     self.bigquery_table = constants.BIGQUERY_TABLE
 
-    if is_running_locally:
+    if constants.IS_DEV_SERVER:
       self.bigquery_dataset = constants.BIGQUERY_DATASET_LOCAL
     else:
       self.bigquery_dataset = constants.BIGQUERY_DATASET_PROD
 
     # Attempt to initialize a connection to BigQuery.
-    self.bigquery = None
-    if is_running_locally:
-      # Local instances require a 'secrets.json' file.
-      secrets_path = os.path.join(os.path.dirname(__file__), 'secrets.json')
-      if (os.path.exists(secrets_path)):
-        with open(secrets_path) as f:
-          auth = json.load(f)
-          credentials = oauth2client.client.SignedJwtAssertionCredentials(
-              auth['client_email'], auth['private_key'],
-              constants.BIGQUERY_URL)
-          self.bigquery = self._build_bigquery_object(credentials)
-      else:
-        logging.warning(
-            'No credentials provided for BigQuery. Logging disabled.')
-    else:
-      # Use the GAE service credentials.
-      credentials = oauth2client.appengine.AppAssertionCredentials(
-          scope=constants.BIGQUERY_URL)
-      self.bigquery = self._build_bigquery_object(credentials)
-
-  def _build_bigquery_object(self, credentials):
-    http = credentials.authorize(httplib2.Http())
-    return discovery.build("bigquery", "v2", http=http)
+    self.bigquery = self._build_bigquery_object()
+    if self.bigquery is None:
+      logging.warning('Unable to build BigQuery API object. Logging disabled.')
+
+  def _build_bigquery_object(self):
+    return googleapi.build(scope=constants.BIGQUERY_URL,
```

Comment:
nit: googleapi may be confusing since it sounds like a google provided module,  like google.appengnie.api. How about a more specific name, like authutils?